src/test/java/com/ksm/wstbackoffice/controller/CountryControllerIT.java

```diff
@@ -54,6 +75,31 @@ public void findAllTest() {
             get("countries").
         then().
             statusCode(200).
-            body("size()", equalTo(5));
+            body("size()", equalTo(21));
+    }
+
+    @ParameterizedTest
+    @MethodSource("findAllByTestArguments")
+    public void findAllByTest(List<String> nameStartsWithFilters, int expectedStatus) {
+        given().
+            queryParam("nameStartsWithFilters", nameStartsWithFilters).
+        when().
+            get("countries/filter-name").
+        then().
+            statusCode(expectedStatus);
     }
+
+    @ParameterizedTest
+    @MethodSource("findAllByFilterNameTestArguments")
+    public void findAllByTest(String nameStartsWithFilters, List<String> expectedData, int expectedStatus) {
+        given().
+            queryParam("nameStartsWithFilters", nameStartsWithFilters).
+        when().
+            get("countries/filter-name").
+        then().
+            statusCode(expectedStatus).
+            body("total.".concat(nameStartsWithFilters), equalTo(expectedData.size())).
+            body("details.".concat(nameStartsWithFilters), hasItems(expectedData.toArray()));
```

Comment:
Instead of hasItems use the one that checks only those countries are there and nothing more. I think something like containsExactlyInAnyOrder should exist

Reply:
done